packages/grafana-ui/src/components/Logs/LogDetailsRow.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { PureComponent } from 'react';
+import { LogRowModel, LogsParser, LogLabelStatsModel, calculateFieldStats } from '@grafana/data';
+
+import { Themeable } from '../../types/theme';
+import { withTheme } from '../../themes/index';
+import { getLogRowStyles } from './getLogRowStyles';
+
+//Components
+import { LogLabelStats } from './LogLabelStats';
+
+interface Props extends Themeable {
+  parsedValue: string;
+  parsedKey: string;
+  field?: string;
+  row?: LogRowModel;
+  canShowMetrics?: boolean;
+  canFilter?: boolean;
+  canFilterOut?: boolean;
+  parser?: LogsParser;
+  onClickFilterLabel?: (key: string, value: string) => void;
+  onClickFilterOutLabel?: (key: string, value: string) => void;
+  createStatsForLogs?: (field: string) => void;
+  getRows?: () => LogRowModel[];
```

Comment:
This is taking lots of optional props which if I understand are optional only in the sense that they are not needed if `canShowMetrics = false` but are needed if `canShowMetrics = true`. This is then a bit hard to understand from the typing, what is needed when as it is not possible to define it in the types itself easily. One way to refactor this would be to have something like this:
```
<LogDetailsRow {...props} renderOnStats={() => <LogLabelStats {...statsProps}/>}/>
```

I think that way we would have 2 components with clear props requirements while `LogDetailsRow` would have only one optional prop `renderOnStats` which would decide on whether any stats can be shown and would be responsible to render them.

Reply:
Yes, I totally agree. I wanted to kind of unify the LogDetailsRow, but it resulted in hard-to-understand and lots-of-optional-props component. So I will split this up and follow your advice with ```<LogDetailsRow {...props} renderOnStats={() => <LogLabelStats {...statsProp```
@torkelo this is why parser is optional -> it is always there and used for the cases, when I used LogDetailsRow with parsed fields. But with labels, we don't use it and it is not required and not used. As mentioned above, I will refactor this and split it into 2 components that have clear props requirements and clear responsibilities.